### ISSUE TITLE
I don't get the label.txt when training the model.txt which causes me to fail to convert to tflite

### ISSUE THREAD

**Issue by Cranky-cat** [tensorflow/tensorflow#59968]
D:\Anaconda\lib\site-packages\tensorflow_lite_support\metadata\python\metadata.py:395: UserWarning: File, 'label.txt', does not exist in the metadata. But packing it to tflite model is still allowed.
  warnings.warn(

from tflite_support.metadata_writers import object_detector
from tflite_support.metadata_writers import writer_utils

ObjectDetectorWriter = object_detector.MetadataWriter

_MODEL_PATH = "D:/android files/8_Car2022_Example_v1.1/app/src/main/assets/test2_mobilenet.tflite"
# Task Library expects label files that are in the same format as the one below.
_LABEL_FILE = "D:/android files/8_Car2022_Example_v1.1/app/src/main/assets/label.txt"
_SAVE_TO_PATH = "D:/android files/8_Car2022_Example_v1.1/app/src/main/assets/test_mobilenet_meta.tflite"
# Normalization parameters is required when reprocessing the image. It is
# optional if the image pixel values are in range of [0, 255] and the input
# tensor is quantized to uint8. See the introduction for normalization and
# quantization parameters below for more details.
# https://www.tensorflow.org/lite/models/convert/metadata#normalization_and_quantization_parameters
_INPUT_NORM_MEAN = 127.5
_INPUT_NORM_STD = 127.5

# Create the metadata writer.
writer = ObjectDetectorWriter.create_for_inference(
    writer_utils.load_file(_MODEL_PATH), [_INPUT_NORM_MEAN], [_INPUT_NORM_STD],
    [_LABEL_FILE])

# Verify the metadata generated by metadata writer.
print(writer.get_metadata_json())

# Populate the metadata into the model.
writer_utils.save_file(writer.populate(), _SAVE_TO_PATH)

**Comment by tilakrayal** [comment]
@Cranky-cat,
Could you please provide the complete code and the dependencies to reproduce the issue in the effective way and helps us to debug the issue and also provide the tensorflow version you are using. Thank you!

**Comment by Cranky-cat** [comment]
`import tensorflow as tf
import matplotlib.pyplot as plt
from time import *


# The dataset loading function indicates the location of the dataset and uniformly processes it to the size of imgheight*imgwidth, while setting batch
def data_load(data_dir, test_data_dir, img_height, img_width, batch_size):
    # Load the training set
    train_ds = tf.keras.preprocessing.image_dataset_from_directory(
        data_dir,
        label_mode='categorical',
        seed=123,
        image_size=(img_height, img_width),
        batch_size=batch_size)
    # Load the test set
    val_ds = tf.keras.preprocessing.image_dataset_from_directory(
        test_data_dir,
        label_mode='categorical',
        seed=123,
        image_size=(img_height, img_width),
        batch_size=batch_size)
    class_names = train_ds.class_names
    # Returns the processed training set, validation set, and class name
    return train_ds, val_ds, class_names


# Build the MobileNet model
# # Model loading, specify the size of image processing and whether to perform transfer learning
def model_load(IMG_SHAPE=(224, 224, 3), class_num=12):
    # Normalization is not required during fine-tuning
    # Load pre-trained mobilenet models
    base_model = tf.keras.applications.MobileNetV2(input_shape=IMG_SHAPE,
                                                   include_top=False,
                                                   weights='imagenet')
    # Freeze the model's backbone parameters
    base_model.trainable = False
    model = tf.keras.models.Sequential([
        # Normalized processing
        tf.keras.layers.experimental.preprocessing.Rescaling(1. / 127.5, offset=-1, input_shape=IMG_SHAPE),
        # Set up the skeleton model
        base_model,
        # Global average pooling of the output of the spine model
        tf.keras.layers.GlobalAveragePooling2D(),
        # 通过全连接层映射到最后的分类数目上
        tf.keras.layers.Dense(class_num, activation='softmax')
    ])
    model.summary()
    # The optimizer trained by the model is the ADAM optimizer, and the loss function of the model is the cross-entropy loss function
    model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
    return model


# Show the curve of the training process
def show_loss_acc(history):
    # Extract the accuracy information and error information of the model training set and the validation set from the history
    acc = history.history['accuracy']
    val_acc = history.history['val_accuracy']
    loss = history.history['loss']
    val_loss = history.history['val_loss']

    # Output the drawing according to the upper and lower structure
    plt.figure(figsize=(8, 8))
    plt.subplot(2, 1, 1)
    plt.plot(acc, label='Training Accuracy')
    plt.plot(val_acc, label='Validation Accuracy')
    plt.legend(loc='lower right')
    plt.ylabel('Accuracy')
    plt.ylim([min(plt.ylim()), 1])
    plt.title('Training and Validation Accuracy')

    plt.subplot(2, 1, 2)
    plt.plot(loss, label='Training Loss')
    plt.plot(val_loss, label='Validation Loss')
    plt.legend(loc='upper right')
    plt.ylabel('Cross Entropy')
    plt.title('Training and Validation Loss')
    plt.xlabel('epoch')
    plt.savefig('results/results_mobilenet.png', dpi=100)


def train(epochs):
    # Start training and record the start time
    begin_time = time()
    # todo Load the dataset, modifying the path to your dataset
    train_ds, val_ds, class_names = data_load("D:/Sign/train",
                                              "D:/Sign/test", 224, 224, 16)
    print(class_names)
    # Load the model
    model = model_load(class_num=len(class_names))
    # Indicate the number of rounds of training epoch and start training
    history = model.fit(train_ds, validation_data=val_ds, epochs=epochs)
    # todo Save Model, modify to the name of the model you want to save
    model.save("models/test_mobilenet")
    # Record the end time
    end_time = time()
    run_time = end_time - begin_time
    print('The cyclic program run time：', run_time, "s")  # The cyclic program run time： 1.4201874732
    # Graph the model training process
    show_loss_acc(history)


if __name__ == '__main__':
    train(epochs=10)
`

**Comment by Cranky-cat** [comment]
import tensorflow as tf
import matplotlib.pyplot as plt
from time import *


# The dataset loading function indicates the location of the dataset and uniformly processes it to the size of imgheight*imgwidth, while setting batch
def data_load(data_dir, test_data_dir, img_height, img_width, batch_size):
    # Load the training set
    train_ds = tf.keras.preprocessing.image_dataset_from_directory(
        data_dir,
        label_mode='categorical',
        seed=123,
        image_size=(img_height, img_width),
        batch_size=batch_size)
    # Load the test set
    val_ds = tf.keras.preprocessing.image_dataset_from_directory(
        test_data_dir,
        label_mode='categorical',
        seed=123,
        image_size=(img_height, img_width),
        batch_size=batch_size)
    class_names = train_ds.class_names
    # Returns the processed training set, validation set, and class name
    return train_ds, val_ds, class_names


# Build the MobileNet model
# # Model loading, specify the size of image processing and whether to perform transfer learning
def model_load(IMG_SHAPE=(224, 224, 3), class_num=12):
    # Normalization is not required during fine-tuning
    # Load pre-trained mobilenet models
    base_model = tf.keras.applications.MobileNetV2(input_shape=IMG_SHAPE,
                                                   include_top=False,
                                                   weights='imagenet')
    # Freeze the model's backbone parameters
    base_model.trainable = False
    model = tf.keras.models.Sequential([
        # Normalized processing
        tf.keras.layers.experimental.preprocessing.Rescaling(1. / 127.5, offset=-1, input_shape=IMG_SHAPE),
        # Set up the skeleton model
        base_model,
        # Global average pooling of the output of the spine model
        tf.keras.layers.GlobalAveragePooling2D(),
        # Mapping to the final number of classifications through a fully connected layer
        tf.keras.layers.Dense(class_num, activation='softmax')
    ])
    model.summary()
    # The optimizer trained by the model is the ADAM optimizer, and the loss function of the model is the cross-entropy loss function
    model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
    return model


# Show the curve of the training process
def show_loss_acc(history):
    # Extract the accuracy information and error information of the model training set and the validation set from the history
    acc = history.history['accuracy']
    val_acc = history.history['val_accuracy']
    loss = history.history['loss']
    val_loss = history.history['val_loss']

    # Output the drawing according to the upper and lower structure
    plt.figure(figsize=(8, 8))
    plt.subplot(2, 1, 1)
    plt.plot(acc, label='Training Accuracy')
    plt.plot(val_acc, label='Validation Accuracy')
    plt.legend(loc='lower right')
    plt.ylabel('Accuracy')
    plt.ylim([min(plt.ylim()), 1])
    plt.title('Training and Validation Accuracy')

    plt.subplot(2, 1, 2)
    plt.plot(loss, label='Training Loss')
    plt.plot(val_loss, label='Validation Loss')
    plt.legend(loc='upper right')
    plt.ylabel('Cross Entropy')
    plt.title('Training and Validation Loss')
    plt.xlabel('epoch')
    plt.savefig('results/results_mobilenet.png', dpi=100)


def train(epochs):
    # Start training and record the start time
    begin_time = time()
    # todo Load the dataset, modifying the path to your dataset
    train_ds, val_ds, class_names = data_load("D:/Sign/train",
                                              "D:/Sign/test", 224, 224, 16)
    print(class_names)
    # Load the model
    model = model_load(class_num=len(class_names))
    # Indicate the number of rounds of training epoch and start training
    history = model.fit(train_ds, validation_data=val_ds, epochs=epochs)
    # todo Save Model, modify to the name of the model you want to save
    model.save("models/test_mobilenet")
    # Record the end time
    end_time = time()
    run_time = end_time - begin_time
    print('The cyclic program run time：', run_time, "s")  # The cyclic program run time： 1.4201874732
    # Graph the model training process
    show_loss_acc(history)


if __name__ == '__main__':
    train(epochs=10)

**Comment by Cranky-cat** [comment]
[train_mobilenet.zip](https://github.com/tensorflow/tensorflow/files/10966111/train_mobilenet.zip)
I used this py to train the model, and converting the trained H5 model file to a tflite model file caused the loss of metadata

**Comment by Cranky-cat** [comment]
@tilakrayal [test.zip](https://github.com/tensorflow/tensorflow/files/10966144/test.zip)
This is the py of converting the file format I am using the tensorflow 2-11 version

**Comment by Cranky-cat** [comment]
I convert file format without label .txt file, so my conversion will fail.
So I would like to ask how to get the label .txt file of the model.

**Comment by Cranky-cat** [comment]
@tilakrayal I can train the model file, but I need the model file to be used on Android, so I need to convert to tflite format.
Thank you for your reply

**Comment by pjpratik** [comment]
Hi @Cranky-cat 

I have observed that in https://github.com/tensorflow/tensorflow/files/10966111/train_mobilenet.zip, the model conversion into tflite is not done.

Also, please refer to the model compatability [requirements](https://www.tensorflow.org/lite/inference_with_metadata/task_library/object_detector#model_compatibility_requirements), which has example lable.txt for pretrained mobile net model. You can create a similar one for your case with class names according to your training dataset.

I have tried metadata tutorial for [object detector](https://www.tensorflow.org/lite/models/convert/metadata_writer_tutorial#object_detectors) and I was able to run without any error. Please find the gist [here](https://colab.research.google.com/gist/pjpratik/841aaa2471d1e75deafd253266c0be20/59968.ipynb). 

Thanks.
